backend/homework/2/src/main/java/org/example/SentimentAnalyzer.java

```diff
@@ -0,0 +1,105 @@
+package org.example;
+import java.util.Arrays;
+public class SentimentAnalyzer {
+    public static int[] detectProsAndCons(String review, String[][] featureSet, String[] posOpinionWords, String[] negOpinionWords) {
+        int[] featureOpinions = new int[featureSet.length];
+        review = review.toLowerCase();
+
+        for (int i = 0; i < featureSet.length; i++) {
+            boolean findFlag = false;
+
+            for (int j = 0; j < featureSet[i].length; j++) {
+                if (review.contains(featureSet[i][j])) {
+                    findFlag = true;
+                    int opinion = getOpinionOnFeature(review, featureSet[i][j], posOpinionWords, negOpinionWords);
+                    featureOpinions[i] = opinion;
+                    break;
+                }
+            }
+
+            if (!findFlag) {
+                featureOpinions[i] = 0;
+            }
+        }
+
+        return featureOpinions;
+    }
+
+    public static int getOpinionOnFeature(String review, String feature, String[] posOpinionWords, String[] negOpinionWords) {
+            int opinion = 0;
+
+            int wasOpinion = checkForWasPhrasePattern(review, feature, posOpinionWords, negOpinionWords);
+            if (wasOpinion != 0) {
+                opinion = wasOpinion;
+            }
+
+            int opinionFirst = checkForOpinionFirstPattern(review, feature, posOpinionWords, negOpinionWords);
+            if (opinionFirst != 0) {
+                opinion = opinionFirst;
+            }
+
+
+        return opinion;
+    }
+
+    private static int checkForWasPhrasePattern(String review, String feature, String[] posOpinionWords, String[] negOpinionWords) {
```

Comment:
try making it modular by refactoring it into multiple methods and classes too whereever possible. Try following singularity principle from SOLIDs.